### PR TITLE
Feedback on Conclusion PR

### PR DESCRIPTION
**Pull request by robinpiets**
This PR is all about the feedback we got from @afjlambert. I've decided to fix this in a separate PR. It includes refactors, code improvements, tests updates and comments.

Please make sure:
- [x] to add a label
- [x] to add one or more reviewers
- [x] you followed our checklist for [functional testing](https://github.com/Amsterdam/vergunningcheck/blob/master/TESTING.md)
- [x] you added the necessary documentation (either in the markdown files or inline)
- [x] you added the necessary automated tests

### CODE REVIEW

File: apps/client/src/components/Conclusion/ConclusionOutcome.test.js [hunk 1]
```diff
@@ -14,40 +14,55 @@ Object.defineProperty(window, "matchMedia", matchMedia);
...
-      needsPermit={false}
-      contactConclusion={{
-        title: "Test",
-        description: "Je moet contact opnemen met amsterdam",
-      }}
-      matomoTrackEvent={() => {}}
-    />
-  );
-  expect(getByTestId(NEED_CONTACT)).toBeTruthy();
+    // Shouldn't be in document
```
Comment (svenjens) [inline apps/client/src/components/Conclusion/ConclusionOutcome.test.js:60]: 🙌

File: apps/client/src/utils/index.js [hunk 2]
```diff
@@ -2,6 +2,11 @@ import isBoolean from "lodash.isboolean";
 import isNumber from "lodash.isnumber";
 import isString from "lodash.isstring";
 
+import { topics } from "../config";
+
+// Find a topic by the slug
+export const findTopicBySlug = (slug) => topics.find((t) => t.slug === slug);
```
Comment (svenjens) [inline apps/client/src/utils/index.js:8]: aren't we using this in the withTracking and withTopic too?

Comment (robinpiets) [reply apps/client/src/utils/index.js:8]: For now I will refactor that too, but that all is going to be ~refactored~ updated soon.

**Review by svenjens — APPROVED** [review]

File: apps/client/src/components/Conclusion/ConclusionOutcome.test.js [hunk 3]
```diff
@@ -14,40 +14,55 @@ Object.defineProperty(window, "matchMedia", matchMedia);
...
 afterEach(cleanup);
 
-it("renders more info for permit description", () => {
-  const { getByTestId } = render(
-    <ConclusionOutcome needsPermit={true} matomoTrackEvent={() => {}} />
-  );
-  expect(getByTestId(NEED_PERMIT)).toBeTruthy();
-  expect(getByTestId(NEED_PERMIT_BUTTON)).toBeTruthy();
-});
+describe("ConclusionOutcome", () => {
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/ConclusionOutcome.test.js:22]: nice

File: apps/client/src/hoc/withTopic.js [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+import { findTopicBySlug } from "../utils";
 
 const withTopic = (Component) => (props) => {
   const sessionContext = useContext(SessionContext);
   const checkerContext = useContext(CheckerContext);
   const { slug } = useParams();
   const { search } = useLocation();
 
-  const topic = topics.find((t) => t.slug === slug);
+  const topic = findTopicBySlug(slug);
```
Comment (afjlambert) [inline apps/client/src/hoc/withTopic.js:17]: 👏

File: apps/client/src/components/Conclusion/NewCheckerModal.tsx [hunk 5]
```diff
@@ -81,13 +82,15 @@ const NewCheckerModal: React.FC<{
           <Label htmlFor="address-input-1" label="Ja">
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/NewCheckerModal.tsx:82]: You could re-use the data-testid here.

**Review by afjlambert — APPROVED** [review]

File: apps/client/src/components/Conclusion/NewCheckerModal.tsx (hunk 5 above)
Comment (robinpiets) [reply apps/client/src/components/Conclusion/NewCheckerModal.tsx:82]: We could do that, yes.

**Pull request merged**